### DISCUSSION TITLE
Studying temperature contour in an annulus geometry: help me please

### DISCUSSION THREAD

**Discussion by amnchacon** [firedrakeproject/firedrake#3349] [Firedrake support]
Good morning.
I want to study what the title of this discussion says, but I'm having a problem that I'm not getting find the solution. After runing appears "Provided solution is a 'Argument', not a Function". How can I solve this issue?  Here is the code:

**Comment by dham** [comment]
You're writing a non-linear variational problem, so there shouldn't be a `TrialFunction`. Those are only used for linear problems. Replace:
```python
T = TrialFunction(V)
```
with
```python
T = Function(V)
```

By the way [please don't post screenshots of text.](https://object-oriented-python.github.io/a1_help.html#avoid-screenshots-of-text)

**Reply by amnchacon** [reply]
Oh, I see @dham . I already did that changing and it worked. I retired the print. Another question: for future questions, how can I put the code just for you (or for the firedrake specialists) and not for everyone?

**Reply by dham** [reply]
There's no need to (or way to) keep the code private. Just put it inside triple backquotes:
```python
    ```python
    print("hello, world!")
    ```
```

**Reply by amnchacon** [reply]
Drs @dham, @colinjcotter , how can I define the boundary conditions in an annulus mesh? The manual says that "the label on_boundary only refers to the side boundaries that take their labels from the base mesh, and not the top or bottom boundaries." There are other labels to define, but I'm trying and it's not being applied on the boundaries.... and, on the other hand, If I have a cylindrus geometry, how I would define the boundary conditions in this case? Thanks ....

**Comment by colinjcotter** [comment]
(I think you meant that there shouldn't be a TrialFunction.)

________________________________________
